pixy/httpapi.go

```diff
@@ -0,0 +1,131 @@
+package pixy
+
+import (
+	"fmt"
+	"net"
+	"net/http"
+	"strconv"
+
+	"github.com/mailgun/kafka-pixy/Godeps/_workspace/src/github.com/gorilla/mux"
+	"github.com/mailgun/kafka-pixy/Godeps/_workspace/src/github.com/mailgun/manners"
+)
+
+const (
+	NetworkTCP  = "tcp"
+	NetworkUnix = "unix"
+
+	hContentLength = "Content-Length"
+
+	fTopicName  = "topic"
+	fRoutingKey = "key"
+)
+
+type HTTPAPIServer struct {
+	addr        string
+	listener    net.Listener
+	httpServer  *manners.GracefulServer
+	kafkaClient IKafkaClient
+	errorCh     chan error
+}
+
+// SpawnHTTPAPIServer starts an HTTP server instance that accepts API requests
+// on the specified network/address and forwards them to the associated
+// Kafka client. The server initialization is performed asynchronously and
+// if it fails then the error is sent down to `HTTPAPIServer.errorCh`.
```

Comment:
Clients can't access `HTTPAPIServer.errorCh`. I would document it as `HTTPAPIServer.ErrorCh()`.